tests/test_scan.py

```diff
@@ -365,3 +365,76 @@ def test_scan_execute_with_soda_server_client_scan_result_does_not_contain_any_e
     )
 
     assert not scan_result.has_errors()
+
+
+def test_scan_measurement_dataclass_fields(
+    scan_definition: str, df: DataFrame
+) -> None:
+    """Valid if the expected fields are present in measurement dataclass is present."""
+    key_list = []
+    scan_result = scan.execute(scan_definition, df)
+    for key in scan_result.measurements[0].__dataclass_fields__:
+        key_list.append(key)
+    assert key_list == ["metric", "column_name", "value", "group_values"]
+
+
+# Identify what should we test for here. The idea is to verify if we have the same set of fields.
+def test_scan_testresult_dataclass_fields(
```

Comment:
as a first test I would not use `execute`. Create a couple `TestResult`, `Measurement` and `Errors` feed them to the `_to_data_frame` functions and validate the output.

Reply:
I have updated the tests as discussed. Please take a look.